File: Codeforces/DeletingDivisors.c
#include <stdio.h>
 
int main()
{
    int t;
    scanf(%d,&t);
 
    while(t--){
        
        int n;
        cin >> n;
        if(n % 2 == 1){
            printf("Bob\n");
            continue;
        }
        int cnt = 0;
        while(n % 2 == 0){
            cnt++;
            n /= 2;
        }
 
        if(n > 1){
            printf("Alice\n");
        }else if(cnt % 2 == 0){
            printf("Alice\n");
        }else printf("Bob\n");
    }
}

File: Codeforces/Deleting_divisors.py
#Deleting Divisors
for _ in range(int(input())):
    n = int(input())
    cnt = 0
    if n%2==1:
        print("Bob")
        continue
    while n%2==0:
        cnt+=1
        n//=2
    if n>1 or cnt%2==0:
        print("Alice")
    else:
        print("Bob")
        
        
#Example:-
#input 
# 4
#1
#4
#12
#69

#output
# Bob
# Alice
# Alice
#;Bob

